packages/shared/src/components/comments/CommentInputOrModal.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactElement } from 'react';
+import {
+  CommentMarkdownInput,
+  CommentMarkdownInputProps,
+} from '../fields/MarkdownInput/CommentMarkdownInput';
+import { ViewSize, useViewSize } from '../../hooks';
+import { LazyModalCommonProps } from '../modals/common/Modal';
+import CommentModal from '../modals/post/CommentModal';
+
+interface CommentInputOrModalProps
+  extends Partial<LazyModalCommonProps>,
+    Omit<CommentMarkdownInputProps, 'className'> {
+  onClose?: () => void;
+  className?: {
+    input?: CommentMarkdownInputProps['className'];
+    modal?: string;
+  };
+  replyToCommentId?: string;
+}
+
+export default function CommentInputOrModal({
+  onClose,
+  className,
+  ...props
+}: CommentInputOrModalProps): ReactElement {
+  const isModal = !useViewSize(ViewSize.Laptop);
+
+  if (isModal) {
+    return <CommentModal {...props} isOpen onRequestClose={onClose} />;
```

Comment:
Need to render modal this way. If we use the `useLazyModal` hook, it does not open the virtual keyboard on Safari on mobile.

Reply:
Good find.

Reply:
Would it be possible to lazy load this through `dynamic` and open the modal with the keyboard on?

Reply:
With `dynamic` it does not open the keyboard reliably. So I think we need to stick with static import. It's just one modal in the end, should not impact the load times too much 🤞